enginetest/enginetests.go

```diff
@@ -4888,10 +4906,12 @@ func checkResults(t *testing.T, require *require.Assertions, expected []sql.Row,
 	}
 
 	// .Equal gives better error messages than .ElementsMatch, so use it when possible
-	if orderBy || len(expected) <= 1 {
-		require.Equal(widenedExpected, widenedRows, "Unexpected result for query %s", q)
-	} else {
-		require.ElementsMatch(widenedExpected, widenedRows, "Unexpected result for query %s", q)
+	if expected != nil {
```

Comment:
This might not be correct (could skip test cases that pass nil for an empty slice). You should instead assert that expected != nil

Reply:
I actually am trying to skip testing the results for the warning case (since I don't know the exact message with the generated commit hashes and am testing the warning message with the new ExpectedWarningMessageSubstring). I checked that only those tests are hitting this code path, but you are totally right that it would be easy for tests to silently not get run by this in the future, so I'll rework this to be more of an explicit opt-out of testing checking results.